lib/evaluation/evaluationClientWrapper.ts

```diff
@@ -95,22 +107,26 @@ export class EvaluationClientWrapper {
     });
     if (selectResults.areRequiredCredentialsPresent === Status.INFO) {
       selectResults.errors = [];
+    } else {
+      selectResults.errors = errors;
+      selectResults.warnings = warnings;
+      selectResults.verifiableCredential = wrappedVerifiableCredentials.map((value) => value.original);
     }
-    return selectResults;
+    return selectResults as SelectResults;
   }
 
   private remapMatches(
-    verifiableCredentials: IVerifiableCredential[],
+    verifiableCredentials: OriginalVerifiableCredential[],
     submissionRequirementMatches?: SubmissionRequirementMatch[],
-    vcsToSend?: IVerifiableCredential[]
+    vcsToSend?: OriginalVerifiableCredential[]
   ) {
     submissionRequirementMatches?.forEach((srm) => {
       if (srm.from_nested) {
         this.remapMatches(verifiableCredentials, srm.from_nested, vcsToSend);
       } else {
         srm.vc_path.forEach((match, index, matches) => {
           const vc = jp.query(verifiableCredentials, match)[0];
-          const newIndex = vcsToSend?.findIndex((svc) => JSON.stringify(svc) === JSON.stringify(vc));
+          const newIndex = vcsToSend?.findIndex((svc) => _.isEqual(svc, vc));
```

Comment:
See remark about introducing a dep. Why was the original code not working? I do not see the need

Reply:
reverted this change